tests/unit/test_solvers/test_casadi_solver.py

```diff
@@ -121,43 +121,43 @@ def test_model_solver_events(self):
         solver = pybamm.CasadiSolver(mode="safe", rtol=1e-8, atol=1e-8)
         t_eval = np.linspace(0, 5, 100)
         solution = solver.solve(model, t_eval)
-        np.testing.assert_array_less(solution.y[0], 1.5)
-        np.testing.assert_array_less(solution.y[-1], 2.5 + 1e-10)
+        np.testing.assert_array_less(solution.y.full()[0], 1.5)
+        np.testing.assert_array_less(solution.y.full()[-1], 2.5 + 1e-10)
         np.testing.assert_array_almost_equal(
-            solution.y[0], np.exp(0.1 * solution.t), decimal=5
+            solution.y.full()[0], np.exp(0.1 * solution.t), decimal=5
         )
         np.testing.assert_array_almost_equal(
-            solution.y[-1], 2 * np.exp(0.1 * solution.t), decimal=5
+            solution.y.full()[-1], 2 * np.exp(0.1 * solution.t), decimal=5
         )
 
         # Solve using "safe" mode with debug off
         pybamm.settings.debug_mode = False
         solver = pybamm.CasadiSolver(mode="safe", rtol=1e-8, atol=1e-8, dt_max=1)
         t_eval = np.linspace(0, 5, 100)
         solution = solver.solve(model, t_eval)
-        np.testing.assert_array_less(solution.y[0], 1.5)
-        np.testing.assert_array_less(solution.y[-1], 2.5 + 1e-10)
+        np.testing.assert_array_less(solution.y.full()[0], 1.5)
```

Comment:
This test fails but I can't understand why it doesn't take the `.full()`.

Reply:
`get_termination_reason` had changed in #1300 and was returning a numpy array